### ISSUE TITLE
Gradle plugin does not support Gradle 6.0

### ISSUE THREAD

**Issue by hjywyj** [greenrobot/greenDAO#1002]
```
Caused by: java.lang.NoSuchMethodError: org.gradle.api.tasks.TaskInputs.property(Ljava/lang/String;Ljava/lang/Object;)Lorg/gradle/api/tasks/TaskInputs;
        at org.greenrobot.greendao.gradle.Greendao3GradlePlugin.createGreendaoTask(Greendao3GradlePlugin.kt:60)
        at org.greenrobot.greendao.gradle.Greendao3GradlePlugin.access$createGreendaoTask(Greendao3GradlePlugin.kt:14)
        at org.greenrobot.greendao.gradle.Greendao3GradlePlugin$apply$1.execute(Greendao3GradlePlugin.kt:47)
        at org.greenrobot.greendao.gradle.Greendao3GradlePlugin$apply$1.execute(Greendao3GradlePlugin.kt:14)
        at org.gradle.configuration.internal.DefaultListenerBuildOperationDecorator$BuildOperationEmittingAction$1.lambda$run$0(DefaultListenerBuildOperationDecorator.java:152)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.reapply(DefaultUserCodeApplicationContext.java:60)
        at org.gradle.configuration.internal.DefaultListenerBuildOperationDecorator$BuildOperationEmittingAction$1.run(DefaultListenerBuildOperationDecorator.java:152)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)

```

**Comment by dandar3** [comment]
The source for `'org.greenrobot:greendao-gradle-plugin' is not available (see #412). 

Will try to decompile and see if I can provide a fix to them, hopefully they will patch and release a new version of the plugin without having to make the sources available.

**Comment by woying456** [comment]
i have the same problem.
@greenrobot @greenrobot-team

**Comment by Deloszhw** [comment]
hit the same issue

**Comment by greenrobot-team** [comment]
Thanks for reporting! This is due to the Gradle plugin not supporting the new Tasks API. This was fixed internally, but there was no time for a release, yet.

TL;DR: the greenDAO Gradle plugin currently does not support Gradle 6.x.

**Comment by greenrobot-team** [comment]
Update: I jumped to conclusions here before properly reading the stack trace. I can not verify this using a minimal example project. https://github.com/greenrobot-team/greenrobot-examples/tree/master/greendao

The `TaskInputs.property(name, value)` method that supposedly does not exist should exist in Gradle 6. https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/TaskInputs.html#property-java.lang.String-java.lang.Object-

Does anyone have a minimal project that reproduces this issue?

Also on an unrelated note: the Android Gradle Plugin does not yet support Gradle 6.0.

**Comment by greenrobot-team** [comment]
Oh my, sorry again: I tested using the latest development version of the plugin. The released version `3.2.2` indeed has this issue.

So this is fixed in the internal version.

**Comment by dandar3** [comment]
@greenrobot-team Any new Android project created with Android Studio 4.0 Canary should show the problem. The new version of Android Gradle plugin that the new AS requires supports Gradle 6, hence the problem. https://developer.android.com/studio/preview

Will try to publish a sample project to replicate the issue.

**Comment by greenrobot-team** [comment]
@dandar3 No need, I made a mistake and CAN actually reproduce this. It's fixed already internally, as stated originally.

**Comment by Alonexx** [comment]
When will the fixed version be released? After I updated my project to gradle 6.0, everything works fine except greendao. I think more and more developers would encounter the same problem.

**Comment by Zhelyazko** [comment]
We are also blocked by the lack of support of gradle 6.x. Any idea when you'll be able to release the fixed version of GreenDao Gradle plugin?

**Comment by adamchodera** [comment]
@greenrobot-team Could you please publicly release version v3.3.0 listed on your website? https://greenrobot.org/greendao/changelog/

**Comment by chenguangtu** [comment]
@greenrobot-team 
Could you please publicly release version v3.3.0 listed on your website? https://greenrobot.org/greendao/changelog/

**Comment by michal-luszczuk** [comment]
@greenrobot-team 
Hello.
What could we do to get version 3.3.0 released?

**Comment by shashachu** [comment]
@greenrobot-team Chiming in here. Android Gradle Plugin 4.0 requires Gradle 6.0 so any app that uses GreenDAO will be blocked from upgrading AGP without the 3.3.0 release.

**Comment by G00fY2** [comment]
Looks like 3.3.0 is now available.

**Comment by dandar3** [comment]
Shame, it took a bit too long for me, after waiting for 3 months I finally managed to spent a little time and move on to using Room in. Thank you @greenrobot-team for everything all this time! 🏆:clap:

**Comment by greenrobot-team** [comment]
As @G00fY2 noticed, we had finally some time to do the `3.3.0` release.
https://greenrobot.org/greendao/changelog/

**Comment by slott** [comment]
@G00fY2 How did you get this to work ?  I just updated and I'm still getting 

   > org.gradle.api.tasks.TaskInputs.property(Ljava/lang/String;Ljava/lang/Object;)Lorg/gradle/api/tasks/TaskInputs;

even after deleting my cache directory...

**Comment by michal-luszczuk** [comment]
Remember that `greendao` consists of 2 things: plugin and library

You probably forgot about updating one of those two.
1. Library - dependency in your app module `gradle.build` file
![image](https://user-images.githubusercontent.com/9006071/83237610-77c7c000-a195-11ea-9f0f-d08708cac949.png)
2. Pluign - dependency in your project level `gradle.build.file`
![image](https://user-images.githubusercontent.com/9006071/83237670-8f9f4400-a195-11ea-8fa0-1a2cdbba1533.png)

**Comment by slott** [comment]
YES - indeed that was what I missed. Thank you _so_ much for keeping this project alive - its an absolutely amazing library and I wouldn't know what to do without it :)

**Comment by WMcCabe8** [comment]
Upgrading to the new greendao 3.3.0 release worked for me in both of my gradle files!
Thank you!
